src/configurationProvider.ts

```diff
@@ -195,16 +194,115 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
     }
 
-    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption | undefined> {
-        const res = await resolveMainClass(folder ? folder.uri : undefined);
+    private async quickFixLaunchConfig(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration):
+        Promise<IMainClassOption | undefined> {
+        if (!config.mainClass) {
+            return this.chooseMainClass(folder);
+        }
+
+        const validationResponse = await validateLaunchConfig(folder, config.mainClass, config.projectName);
+        if (validationResponse.mainClass.passed && validationResponse.projectName.passed) {
+            return {
+                mainClass: config.mainClass,
+                projectName: config.projectName,
+            };
+        } else {
+            const errors: string[] = [];
+            if (!validationResponse.mainClass.passed) {
+                errors.push(String(validationResponse.mainClass.message));
+            }
+
+            if (!validationResponse.projectName.passed) {
+                errors.push(String(validationResponse.projectName.message));
+            }
+
+            if (validationResponse.proposals && validationResponse.proposals.length) {
+                const answer = await utility.showErrorMessageWithTroubleshooting({
+                    message: errors.join("\n"),
+                    type: Type.USAGEERROR,
+                }, "Quick Fix");
+                if (answer === "Quick Fix") {
+                    const quickFix: IMainClassOption = await this.showMainClassQuickPick(validationResponse.proposals,
+                        "Please select main class<project name>", true);
+                    this.persistLaunchConfig(folder, config, quickFix);
+                    return quickFix;
+                }
+            } else {
+                utility.showErrorMessageWithTroubleshooting({
+                    message: errors.join("\n"),
+                    type: Type.USAGEERROR,
+                });
+            }
+
+            return;
+        }
+    }
+
+    private persistLaunchConfig(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration, change: IMainClassOption): void {
+        if (!change) {
+            return;
+        }
+
+        const launchConfigurations: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("launch", folder);
+        const rawConfigs: vscode.DebugConfiguration[] = launchConfigurations.configurations;
+        const targetIndex: number = this.findDebugConfig(rawConfigs, config);
+        if (targetIndex >= 0) {
+            rawConfigs[targetIndex].mainClass = change.mainClass;
+            rawConfigs[targetIndex].projectName = change.projectName;
+
+            launchConfigurations.update("configurations", rawConfigs);
+        }
+    }
+
+    private findDebugConfig(rawConfigs: vscode.DebugConfiguration[], config: vscode.DebugConfiguration): number {
+        const candidates = [];
```

Comment:
lodash.isEqual does deep comparison.
_.findIndex(rawConfigs, _.isEqual) can get you the index.